src/score_gen/score.rs

```diff
@@ -395,6 +395,16 @@ fn scalar_to_bits(scalar: &Scalar) -> [u8; 256] {
     res
 }
 
+fn log_2_roudup(mut scalar: Scalar) -> u64 {
+    scalar = scalar.reduce();
+    let mut counter = 0u64;
+    while scalar > Scalar::one().reduce() {
```

Comment:
As mentioned in our call, we can optimize this since we're working with the reduce version of the scalar.
Basically we could just checking the MSB in the four `u64` that internally the `Scalar` use. I filed a tech debt #22 for that.